lib/facterdb.rb

```diff
@@ -4,7 +4,35 @@
 module FacterDB
 
   def self.database
-    @database ||= "[#{Dir.glob("#{File.expand_path(File.join(File.dirname(__FILE__), '../facts'))}/*/*.facts").map { |f| File.read(f) }.join(',')}]\n"
+    @database ||= "[#{facterdb_fact_files.map { |f| File.read(f) }.join(',')}]\n"
+  end
+
+  # @return [Array[String]] -  list of all files found in the default facterdb facts path
+  def self.default_fact_files
+    proj_root = File.join(File.dirname(File.dirname(__FILE__)))
+    facts_dir = File.expand_path(File.join(proj_root, 'facts'))
+    Dir.glob(File.join(facts_dir, "**", '*.facts'))
+  end
+
+  # @return [Array[String]] -  list of all files found in the user supplied facterdb facts path
+  # @param fact_paths [String] - a comma separated list of paths to search for fact files
+  def self.external_fact_files(fact_paths = ENV['FACTERDB_SEARCH_PATH'])
+    fact_paths ||= ''
+    return [] if fact_paths.empty?
+    paths = fact_paths.split(',').map do |fact_path|
```

Comment:
Rather than splitting by `,`, can we change this to `File::PATH_SEPARATOR` which will be a bit more idiomatic to users (`:` on Linux/BSD/etc, `;` on Windows)